.github/workflows/cmake.yml

```diff
@@ -53,14 +57,10 @@ jobs:
           sudo apt -y update
           sudo apt -y install ${{ matrix.install }}
 
-      - name: install libsystemd if Ubuntu
-        if: ${{ startsWith(matrix.os, 'ubuntu') }}
-        run: |
-          sudo apt -y update
-          sudo apt -y install libsystemd-dev
-
       - name: configure cmake
-        run: cmake ${{matrix.cmake_opts}} -DCMAKE_TOOLCHAIN_FILE=${{github.workspace}}/toolchains/${{matrix.toolchain}} -S ${{ github.workspace }} -B ${{github.workspace}}/build
+        env:
+          TOOLCHAIN: ${{ matrix.toolchain && format('{0}/toolchains/{1}', github.workspace, matrix.toolchain) || '' }}
```

Comment:
for anyone who might be wondering, the trick I'm using to convince `cmake` to use visual studio is to not provide a toolchain spec at all. `cmake` will figure it out.